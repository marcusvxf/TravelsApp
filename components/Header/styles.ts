import EStyleSheet from 'react-native-extended-stylesheet';

EStyleSheet.build();
export const headerStyle = EStyleSheet.create({
  container: {
    marginTop: '3rem',
    marginBottom: '1rem',
    marginLeft: '5%',
  },
});
