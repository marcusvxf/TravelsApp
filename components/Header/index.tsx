import React from 'react';
import {View, Text} from 'react-native';

import {headerStyle} from './styles';

function Header(props: any) {
  return (
    <View style={headerStyle.container}>
      <Text className="text-3xl font-extrabold text-black ">
        {props.children}
      </Text>
    </View>
  );
}

export default Header;
