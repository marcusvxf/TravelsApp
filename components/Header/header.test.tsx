import {render} from '@testing-library/react-native';
import Header from '.';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Header', () => {
  test('Header Rendered', () => {
    const header = renderer.create(<Header>Proxima pagina</Header>).toJSON();
    expect(header).toMatchSnapshot();
  });
  test('Header Show Correct text', () => {
    const {getByText} = render(<Header>Proximas viagens</Header>);
    const element = getByText('Proximas viagens');
    expect(element).toBeTruthy();
  });
});
