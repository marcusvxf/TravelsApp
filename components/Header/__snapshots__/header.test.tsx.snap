// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header Rendered 1`] = `
<View
  style={
    Object {
      "marginBottom": 16,
      "marginLeft": 16,
      "marginTop": 48,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 30,
          "lineHeight": 36,
        },
        Object {
          "fontWeight": "800",
        },
        Object {
          "color": "#000",
        },
      ]
    }
  >
    Proxima pagina
  </Text>
</View>
`;
